openlibrary/plugins/openlibrary/js/readmore.js

```diff
@@ -55,7 +55,7 @@ export class ReadMoreComponent {
         this.fullHeight = this.$content.scrollHeight;
         // Fudge factor to account for non-significant read/more
         // (e.g missing a bit of padding)
-        if (this.$content.scrollHeight <= (this.collapsedHeight + 1)) {
+        if (this.$content.scrollHeight <= (this.collapsedHeight + (this.$readMoreButton?.offsetHeight | 0) + 1)) {
```

Comment:
Nice this is working almost perfectly ! A few tweaks:

- We'll want to compute and _store_ the offsetHeight of the readMore button the first time reset is called. Otherwise when the read more button is hidden and `reset` is called, it will switch between existing and not existing. (You can observe this by trying to resize the window on testing. You'll see the read more button appear and disappear, since sometimes the height is there and sometimes it's zero). The first time reset is called, the read more is always visible, so you can likely do something like: `this.readMoreHeight ||= this.$readMoreButton.offsetHeight;` at the top of this method.

- Also we don't need the `?.` since it's always defined, and we don't need to default to zero since `offsetHeight` will always be a valid number.
- (also: Small typo, I think you mean `||` instead of `|` 😁 )

Reply:
Thank you for pointing that out—I’ve just fixed it!

Reply:
We'll need to implement that first bullet point as well!